lib/galaxy/webapps/galaxy/config_schema.yml

```diff
@@ -2436,6 +2436,38 @@ mapping:
           When false, the most recently added compatible item in the history will
           be used for each "Set at Runtime" input, independent of others in the workflow.
 
+      simplified_workflow_run_ui:
+        type: str
+        default: 'prefer'
+        enum: ['off', 'prefer']
+        required: false
+        desc: |
+          If set to 'off' by default, always use the traditional workflow form that renders
+          all steps in the GUI and serializes the tool state of all steps during
+          invocation. Set to 'prefer' to default to a simplified workflow UI that
+          only renders the inputs if possible (the workflow must have no disconnected
+          runtime inputs and not replacement parameters within tool steps). In the
+          future 'force' may be added an option for Galaskio-style servers that should
+          only render simplified workflows.
+
+      simplified_workflow_run_ui_target_history:
```

Comment:
This and the setting below really feel like they should be user preferences, not galaxy-wide configuration settings.

Reply:
This project stemmed from a desire to present a completely different, simplified UI around workflows - including one that doesn't even need to present the concept of a history to users. If you wanted to deploy such an app you really wouldn't want to make this a user setting. 

I'm all for decoupling - the corresponding attributes on Vue components do have defaults. They should work stand-alone in other scenarios. So they are decoupled properly in my mind. Just because we don't want to force the developer using the components to define these doesn't mean the admins shouldn't be allowed to pick defaults. Likewise, just because it would likely be good to allow user preferences in the abstract for more typical uses doesn't mean we shouldn't allow admins to be able to control how the app is configured as a default.

Having one part of the UI depend on Galaxy's concept of a user preference tab and the backend for storing and fetching those - is in fact more coupling to my mind than this. Again, the components would work without those so it seems like a fine sort of optional coupling. We have independent ways to configure these things and the component internals don't depend on how those are being configured - that would seem to be the goal to me.

Anyway, while I agree with some of this, adding the user preferences seems like outside the scope of the initial implementation. If you insist on my adding setting I will consider it to get this done, but even in that scenario I still think the defaults should be configurable by the deployer. Though I'm totally open to there being a different way for the deployer to configure the client than Galaxy's YAML.

Reply:
>  If you insist 

@jmchilton I really wasn't insisting on anything, sorry if you thought as much.

I think this component should be seen as a simple , default togglable level of detail view than a different, server-set preference.  In my mind the second pass, the actual workflow-centric interface addition, that's what would add these extra properties (however they end up) and pass them to this view.  This is nice progress as a straightforward level of detail toggle on the existing workflow interface.  That's what I was trying to get at above, with the configuration defaults -- it doesn't serve anyone to force one view over the other with just this piece, I don't think, and I suppose a user could just flip the interface one way or another if they wanted anyway by swapping a bit of client-side data.

Anyway, for now, it's fair enough that until we have a way to configure more (and default) client(s), that this can live in the core galaxy configuration.  If you want this configurable at this point, then it's fine.